framework/src/meshgenerators/BreakMeshByBlockGenerator.C

```diff
@@ -437,3 +538,87 @@ BreakMeshByBlockGenerator::findBlockPairs(subdomain_id_type block_one, subdomain
       return true;
   return false;
 }
+
+void
+BreakMeshByBlockGenerator::syncConnectedBlocks(
+    const std::map<dof_id_type, std::vector<dof_id_type>> & node_to_elem_map,
+    MeshBase & mesh,
+    std::map<dof_id_type, std::set<subdomain_id_type>> & nodeid_to_connected_blocks)
+{
+  // Phase 0: Each rank computes its local connected blocks for nodes it holds
+  for (const auto & map_entry : node_to_elem_map)
+  {
+    std::set<subdomain_id_type> connected_blocks;
+
+    const auto & elem_ids = map_entry.second;
+    for (const dof_id_type elem_id : elem_ids)
+    {
+      const Elem * current_elem = mesh.elem_ptr(elem_id);
+
+      subdomain_id_type block_id = blockRestrictedElementSubdomainID(current_elem);
+
+      nodeid_to_connected_blocks[map_entry.first].insert(block_id);
+    }
+  }
+
+  if (mesh.is_replicated())
+    return; // No synchronization needed for replicated meshes
+
+  auto & comm = mesh.comm();
+  const auto mesh_pid = mesh.processor_id();
+
+  // Phase 1: Ghost nodes push their connected blocks to the owner
+  using NodeConnectedBlocksTuple = std::
+      tuple<dof_id_type, std::vector<subdomain_id_type>, processor_id_type>; // (node_id,
+                                                                             // connected_blocks,
+                                                                             // ghost_pid)
+  std::map<processor_id_type, std::vector<NodeConnectedBlocksTuple>> to_owner;
+  for (const auto & map_entry : nodeid_to_connected_blocks)
+  {
+    const auto node_id = map_entry.first;
+    const Node * node = mesh.node_ptr(node_id);
+    if (node && node->processor_id() != mesh_pid)
+    {
+      const auto & blocks = map_entry.second;
+      to_owner[node->processor_id()].emplace_back(
+          node_id, std::vector<subdomain_id_type>(blocks.begin(), blocks.end()), mesh_pid);
+    }
+  }
+
+  // Owner receives and merges ghost info into its local map, and tracks subscribers
+  std::unordered_map<dof_id_type, std::set<processor_id_type>> subscribers;
+  Parallel::push_parallel_vector_data(
+      comm,
+      to_owner,
+      [&](processor_id_type, const std::vector<NodeConnectedBlocksTuple> & recv_data)
+      {
+        for (const auto & [node_id, blocks_vec, ghost_pid] : recv_data)
+        {
+          subscribers[node_id].insert(ghost_pid);
+          nodeid_to_connected_blocks[node_id].insert(blocks_vec.begin(), blocks_vec.end());
+        }
+      });
+
+  // Phase 2: Node owners broadcast complete connected blocks back to subscribers
+  std::map<processor_id_type, std::vector<NodeConnectedBlocksPair>> from_owner;
+  for (const auto & [node_id, sub_pids] : subscribers)
+  {
+    const Node * node = mesh.node_ptr(node_id);
+    if (node && node->processor_id() == mesh_pid)
+    {
+      const auto & blocks_set = nodeid_to_connected_blocks.at(node_id);
```

Comment:
```suggestion
      const auto & blocks_set = libmesh_map_find(nodeid_to_connected_blocks, node_id);
```
In general we should never be using `map::at` because if it fails it doesn't give users any context. And oftentimes if we're using `at` it means we may have missed an opportunity to create and re-use an iterator